crates/bevy_ui/src/render/mod.rs

```diff
@@ -522,125 +529,72 @@ pub fn extract_uinode_borders(
         .map(|r| r * ui_scale.0);
 
         let border_radius = clamp_radius(border_radius, uinode.size(), border.into());
-        let transform = global_transform.compute_matrix();
 
-        extracted_uinodes.uinodes.insert(
-            commands.spawn_empty().id(),
-            ExtractedUiNode {
-                stack_index: uinode.stack_index,
-                // This translates the uinode's transform to the center of the current border rectangle
-                transform,
-                color: border_color.0.into(),
-                rect: Rect {
-                    max: uinode.size(),
-                    ..Default::default()
-                },
-                image,
-                atlas_scaling: None,
-                clip: clip.map(|clip| clip.clip),
-                flip_x: false,
-                flip_y: false,
-                camera_entity,
-                border_radius,
-                border,
-                node_type: NodeType::Border,
-            },
-        );
-    }
-}
-
-pub fn extract_uinode_outlines(
-    mut commands: Commands,
-    mut extracted_uinodes: ResMut<ExtractedUiNodes>,
-    default_ui_camera: Extract<DefaultUiCamera>,
-    uinode_query: Extract<
-        Query<(
-            &Node,
-            &GlobalTransform,
-            &ViewVisibility,
-            Option<&CalculatedClip>,
-            Option<&TargetCamera>,
-            &Outline,
-        )>,
-    >,
-) {
-    let image = AssetId::<Image>::default();
-    for (node, global_transform, view_visibility, maybe_clip, camera, outline) in &uinode_query {
-        let Some(camera_entity) = camera.map(TargetCamera::entity).or(default_ui_camera.get())
-        else {
-            continue;
-        };
-
-        // Skip invisible outlines
-        if !view_visibility.get()
-            || outline.color.is_fully_transparent()
-            || node.outline_width == 0.
+        // don't extract border if no border or the node is zero-sized (a zero sized node can have an outline).
+        if uinode.size().x > 0.
```

Comment:
Any reason this is no longer an early continue? I'd prefer to keep the indentation low.